### ISSUE TITLE
JPEG/JFIF: Incorrect pixel aspect ratio stored in output file header.

### ISSUE THREAD

**Issue by CybeatB** [OpenImageIO/oiio#1999]
### Problem

I attempted to modify the pixel aspect ratio of a JPEG/JFIF image with `oiiotool`.

**Expected behavior:**
I expected the output image to correctly encode its new pixel aspect ratio in the X_Density & Y_Density fields, as per the spec.

**Actual behavior:**
The pixel aspect ratio changed, but the value encoded in the header didn't match the new image resolution.

### Steps to Reproduce

1. I had an input image from some work I was doing, but it's easier to see what's going on by creating a synthetic test case:
   ```
   oiiotool --create 480x320 3 --attrib XResolution 65535 --attrib YResolution 65535 -o example.jpg
   ```
   The dimensions of the image don't really matter, the important part is setting the XResolution and YResolution attributes to the maximum value that we can store in the relevant JFIF header fields.
2. Modify the pixel aspect ratio of the image:
   ```
   oiiotool example.jpg --pixelaspect 2 -o output.jpg
   ```
3. Hexdump the first ~20 bytes of the modified image so that we can examine the JFIF header.
   ```
   ff d8 ff e0 00 10 4a 46 49 46 00 01 01 00 ff ff ff fe 00 00 ...
   ---------------------------------------------------------------
   0           4           8           12          16          20
   ```
   We see that the density fields are `00 ff ff ff fe`, giving us an X_Density of 65535 and a Y_Density of 65534 with no units.
   This prevents the image from displaying correctly in RV and Nuke.

I've had a poke through the code, and it looks like the YResolution value is being doubled by the `--pixelaspect 2` operation, then truncated to its lower 2 bytes to fit into the Y_Density field.

My first thought would be to normalise XResolution & YResolution before setting X_Density & Y_Density whenever the units are 'None'. If I can make some time later, I'll give it a go and make a pull request.

For what I was attempting when I discovered the issue, this is a suitable work-around:
```oiiotool input.jpg --resize WxH --attrib XResolution 2 --attrib YResolution 1 -o output.jpg```

### Versions

* OIIO branch/version: 1.6.17
* OS: CentOS 6
* C++ compiler: g++ 4.8.5

**Edit:** Fixed a typo.

**Comment by lgritz** [comment]
I see, yeah. The JPEG output should probably check for density values that don't fit in 16 bits and automatically scale them down to preserve aspect ratio but still fit in the ushort data type. I'll see if I can make a quick patch for that in modern releases, but 1.6 is much too old to backport to.

**Comment by lgritz** [comment]
Proposed fix in #2002
